src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,290 @@
+     */
+    public void updateTasksFile() {
+        try {
+            File writeFile = new File(FILE_PATH);
+            if (!writeFile.exists()) {
+                System.out.println("Creating new file 'duke.txt':");
+                writeFile.createNewFile();
+            }
+
+            FileWriter writer = new FileWriter(writeFile);
+            for (Task task : tasks) {
+                try {
+                    String writeLine = parser.convertToFileInput(task);
+                    writer.write(writeLine + BREAK);
+                } catch (ParseException pe) {
+                    System.out.println(pe.getMessage());
+                }
+            }
+            writer.close();
+        } catch (IOException ioe) {
+            System.out.println(ioe.getMessage());
+        }
+    }
+
+    /**
+     * Adds a Task object to the ArrayList of Tasks. Each addition is immediately written to saved file.
+     *
+     * @param task Task to be added to list
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+        updateTasksFile();
+    }
+
+    /**
+     * Creates and returns a specific Task object added to the TaskList. Each additiion is immediately
+     * written to saved file.
+     *
+     * Depending on the task type passed in, a ToDo, Deadline, or Event object is created, then added to the TaskList.
+     *
+     * @param taskType type of Task, i.e., ToDo, Event, Deadline
+     * @param userInputLine full input line entered by user
+     * @return added Task object
+     * @throws InvalidCommandException
+     */
+    public Task addTask(String taskType, String userInputLine) throws InvalidCommandException {
+        String taskDesc = getDescFromInput(userInputLine);
+
+        Task taskAdded;
+        switch (taskType) {
+        case COMMAND_TODO:
+            taskAdded = new ToDo(taskDesc);
+            break;
+        case COMMAND_DEADLINE:
+            taskAdded = new Deadline(taskDesc);
+            break;
+        case COMMAND_EVENT:
+            taskAdded = new Event(taskDesc);
+            break;
+        default:
+            throw new InvalidCommandException();
+        }
+
+        tasks.add(taskAdded);
+        updateTasksFile();
+
+        return taskAdded;
+    }
+
+    /**
+     * Finds all tasks that have task descriptions matching the search input for this command.
+     *
+     * @param userInputLine full user input line, containing the search term(s)
+     * @return List of Tasks that match the search
+     */
+    public List<Task> find(String userInputLine) {
+        String findDesc = getDescFromInput(userInputLine);
+
+        List<Task> tasksFound = tasks.stream()
+                .filter(task -> task.getTaskDesc().contains(findDesc))
+                .collect(Collectors.toList());
+
+        return tasksFound;
+    }
+
+    private String getDescFromInput(String userInputLine) {
+        String taskDesc = "";
+        String[] taskDetails = userInputLine.split(" ", 2);
```

Comment:
Just like the above comments, can consider making a constant for these magic literals '1' and '2'. Perhaps can think along the lines of, what do they mean?

Reply:
I didn't change the '1' and '2' here because I personally feel it will be a little excessive, but I have included descriptive comments to address the "WHAT" and "WHY" for these two lines in particular.